### ISSUE TITLE
Compilation of test specs with UTF-8-strings fail on Windows.

### ISSUE THREAD

**Issue by tschoening** [kaitai-io/kaitai_struct_tests#16]
Some of your files are UTF-8 encoded and contain UTF-8 characters, `javac` under Windows by default assumes that Java files are encoded in `windows-1252`. This can be fixed easily, the interesting question is where: `run-java` contains various invocations of `javac` and currently only `3. Test specs` fails because of encoding issues. So you need to decide if encoding should be added to all invocations or only the problematic currently and if the first, outsource encoding to `config`?

> run-java: compiling test specs
C:\Users\tschoening\Documents\Eclipse\Java Bug 2368\Libs Java 3rd\Parsers\Generators\kaitai_struct\tests\master\src\spec\java\src\io\kaitai\struct\spec\TestStrEncodings.java:14: error: unmappable character for encoding windows-1252
        assertEquals(r.str2(), "Ò?ôÒéôÒ?½Ò?íÒ?»");

    javac -encoding UTF-8 -cp "$JAVA_CLASSES_DIR:$JAVA_TESTNG_JAR" [...]

**Comment by GreyCat** [comment]
Aww, that's true. Actually, `javac` uses some "system default" encoding, and, mysteriously, it thinks that Windows will be fine with having so called "ANSI encoding" (which tends to vary a lot from one system to another due to different language packs installed and "legacy ANSI encoding" setting).

I guess we can safely assume that all our sources in all languages would be UTF-8 (or ASCII, if language wouldn't support UTF-8).

**Comment by GreyCat** [comment]
I've added https://github.com/kaitai-io/kaitai_struct_tests/commit/dc40150da0985e57f66c1441385b70ba477f52da to address this issue. The solution is somewhere in the middle: I've added `-encoding` to all invocations, but kept `JAVAC_OPTS` variable in `run-java`: it's not like there's much to configure there, it's more or less hard-coded stuff. While it makes sense to configure relative locations of runtimes and other dirs with different checkout configurations, test sources would be always in UTF-8, so it's not really something that one would want to configure.
